public/app/features/alerting/unified/components/silences/SilencesDrawer.tsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+
+import { Drawer, Stack } from '@grafana/ui';
+import { AlertManagerPicker } from 'app/features/alerting/unified/components/AlertManagerPicker';
+import { GrafanaAlertmanagerDeliveryWarning } from 'app/features/alerting/unified/components/GrafanaAlertmanagerDeliveryWarning';
+import { SilencesEditor } from 'app/features/alerting/unified/components/silences/SilencesEditor';
+import { getDefaultSilenceFormValues } from 'app/features/alerting/unified/components/silences/utils';
+import { useAlertmanager } from 'app/features/alerting/unified/state/AlertmanagerContext';
+import { isGrafanaRulerRule } from 'app/features/alerting/unified/utils/rules';
+import { CombinedRule } from 'app/types/unified-alerting';
+
+type Props = {
+  rule: CombinedRule;
+  onClose: () => void;
+};
+
+/**
+ * For a given rule, renders a drawer containing silences editor and Alertmanager selection
+ */
+const SilencesDrawer = ({ rule, onClose }: Props) => {
+  const ruleUid = isGrafanaRulerRule(rule.rulerRule) ? rule.rulerRule?.grafana_alert.uid : '';
+
+  const formValues = getDefaultSilenceFormValues();
+  const { selectedAlertmanager } = useAlertmanager();
+
+  return (
+    <Drawer
+      title="Silence alert rule"
+      subtitle="Configure silences to stop notifications from a particular alert rule."
+      onClose={onClose}
+      size="md"
+    >
+      <Stack gap={2} direction={'column'}>
+        <GrafanaAlertmanagerDeliveryWarning currentAlertmanager={selectedAlertmanager!} />
+
+        <AlertManagerPicker />
```

Comment:
As commented above I do wonder if we need the picker or maybe we should only use Grafana built-in AM in that case